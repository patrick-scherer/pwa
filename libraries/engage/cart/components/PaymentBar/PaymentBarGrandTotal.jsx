// @flow
import * as React from 'react';
import { SurroundPortals } from '@shopgate/engage/components';
import {
  CART_PAYMENT_BAR_TOTALS_GRAND_TOTAL,
} from '@shopgate/pwa-common-commerce/cart/constants/Portals';
import CartTotalLine from '@shopgate/pwa-ui-shared/CartTotalLine';
import { CartContext } from '../../cart.context';
import { spacer } from './PaymentBarContent.style';
import connect from './PaymentBarGrandTotal.connector';

type Props = {
  amount: number,
  showSeparator?: boolean,
  label?: string,
  className?: string,
}

/**
 * The GrandTotal component.
 * @returns {JSX}
 */
function PaymentBarGrandTotal({
  amount, label, showSeparator, className,
}: Props) {
  const {
    config: { hideTotal }, isLoading, currency, hasPromotionCoupons,
  } = React.useContext(CartContext);

  if (hideTotal) {
    return null;
  }

  return (
    <SurroundPortals portalName={CART_PAYMENT_BAR_TOTALS_GRAND_TOTAL}>
      <CartTotalLine isDisabled={isLoading} type="grandTotal" className={className}>
        <CartTotalLine.Label label={label} showSeparator={showSeparator} />
        <CartTotalLine.Amount amount={amount} currency={currency} />
        { hasPromotionCoupons && (
          <CartTotalLine.Spacer className={spacer} />
        )}
      </CartTotalLine>
    </SurroundPortals>
  );
}

PaymentBarGrandTotal.defaultProps = {
  className: null,
  showSeparator: true,
  label: 'cart.total',
};

export default connect(PaymentBarGrandTotal);
