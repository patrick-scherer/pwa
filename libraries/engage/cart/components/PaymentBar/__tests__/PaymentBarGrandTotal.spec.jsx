import React from 'react';
import { shallow } from 'enzyme';
import PaymentBarGrandTotal from '../PaymentBarGrandTotal';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: () => ({
    config: { hideTotal: false },
    currency: 'EUR',
    isLoading: false,
  }),
}));

jest.mock('../PaymentBarGrandTotal.connector', () => cmp => cmp);
jest.mock('@shopgate/engage/components');

describe('<PaymentBarGrandTotal />', () => {
  it('should render without issues', () => {
    const wrapper = shallow(<PaymentBarGrandTotal amount={10} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
