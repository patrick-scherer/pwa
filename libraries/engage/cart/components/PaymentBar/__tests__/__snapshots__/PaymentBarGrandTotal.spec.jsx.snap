// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentBarGrandTotal /> should render without issues 1`] = `
<CartTotalLine
  className={null}
  isDisabled={false}
  type="grandTotal"
>
  <Label
    label="cart.total"
    labelParams={Object {}}
    showSeparator={true}
    suffix={null}
  />
  <Amount
    amount={10}
    currency="EUR"
  />
</CartTotalLine>
`;
